### PR TITLE
bitcoin: 0.17.3 -> 0.18.0, enable and run unit tests

### PR DESCRIPTION
**Pull request by dtzWill**
###### Motivation for this change

https://bitcoincore.org/en/releases/0.18.0/

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by Lassulus — APPROVED** [review]

**Review by jonasnick — COMMENTED** [review]
Can confirm that
- the `sha256` hash matches what is signed by the Bitcoin Core maintainer
- this builds on my system
- adding the rapidcheck `check` dependency will result in more tests being run (f.e. `key_properties`).
- without setting  `"LC_ALL=C.UTF-8"` the tests fail with:
```
make  check-TESTS check-local
<snip>
Running 1 test case...
Entering test module "Bitcoin Test Suite"
test/fs_tests.cpp(10): Entering test suite "fs_tests"
test/fs_tests.cpp(12): Entering test case "fsbridge_fstream"
unknown location(0): fatal error: in "fs_tests/fsbridge_fstream": boost::system::system_error: boost::filesystem::path codecvt to string: error
test/fs_tests.cpp(12): last checkpoint: "fsbridge_fstream" test entry
```

Do you have an idea why `LC_ALL=C.UTF-8` must be set? When running the unit tests manually in `bin/test_bitcoin` I also have to set `LC_ALL.UTF-8` or above error will be reproduced. However, running bitcoind and bitcoin-cli seems to work fine (but I didn't do extensive tests).

**Comment by roconnor** [comment]
Can you tell me how to verify the SHA256 hash?

I have is `0ps0vw9iknz1b1sx74rabd1yhlxvwbd0aimjzn9hlqkvw286hkjy` which is presumably the same, but I don't know how to tell.

**Comment by jonasnick** [comment]
@roconnor `$ sha256sum bitcoin-0.18.0.tar.gz`

**Comment by roconnor** [comment]
Ah it is the literal result of `sha256sum`.  Silly me!

File: pkgs/applications/altcoins/bitcoin.nix [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
 
   src = fetchurl {
     urls = [ "https://bitcoincore.org/bin/bitcoin-core-${version}/bitcoin-${version}.tar.gz"
              "https://bitcoin.org/bin/bitcoin-core-${version}/bitcoin-${version}.tar.gz"
            ];
-    sha256 = "0am4pnaf2cisv172jqx6jdpzx770agm8777163lkjbw3ryslymiy";
+    sha256 = "5e4e6890e07b620a93fdb24605dae2bb53e8435b2a93d37558e1db1913df405f";
   };
 
   nativeBuildInputs = [ pkgconfig autoreconfHook ]
```
Comment (roconnor) [inline pkgs/applications/altcoins/bitcoin.nix]: Could you do me a favour and move `python3` from `nativeBuildInputs` to `checkInputs`?  We can also do it in a new PR if you think that is better.

Comment (dtzWill) [reply pkgs/applications/altcoins/bitcoin.nix]: Done! Haven't built it yet but doing so now and seems reasonable and safe to me, but we can see what borg says :).

**Review by roconnor — APPROVED** [review]
Thanks for doing this @dtzWill. Your PR is much better than anything I would have come up with.

**Comment by roconnor** [comment]
@jb55 In case you wish to review.

**Comment by jb55** [comment]
tests pass, running now. seems to be working. lgtm!

**Pull request merged**